index-ml.md

```diff
@@ -99,32 +90,59 @@ Tool | Description | Reference
 {: .table.table-striped .tooltable}
 
+## Dimension reduction
+
+Reducing the dimension of your feature set.
+
+Tool | Description | Reference
+--- | --- | ---
+{% include tool.html id="lda_analy1" %} | Perform LDA Linear Discriminant Analysis|
+{% include tool.html id="pca1" %} | Principal Component Analysis|
+{% include tool.html id="kpca1" %} | Kernel Principal Component Analysis|
+{% include tool.html id="cca1" %} | Canonical Correlation Analysis|
+{% include tool.html id="kcca1" %} | Kernel Canonical Correlation Analysis|
+{: .table.table-striped .tooltable}
 
-## Model selection
+## Model building
 
-Comparing, validating and choosing parameters and models.
+Building general machine learning models.
+
+Tool | Description | Reference
+--- | --- | ---
+{% include tool.html id="Estimator Attributes" %} | Estimator attributes get important attributes from an estimator or scikit object | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
+{% include tool.html id="Stacking Ensemble Models" %} | Stacking Ensembles builds stacking, voting ensemble models with numerous base options | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
+{% include tool.html id="Search CV" %} | Hyperparameter Search performs hyperparameter optimization using various SearchCVs  | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
+{% include tool.html id="Build Pipeline" %} | Pipeline Builder an all-in-one platform to build pipeline, single estimator, preprocessor and custom wrappers | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
+{: .table.table-striped .tooltable}
+
+## Model evaluation
+
+Evaluation, validating and choosing parameters and models.
 
 Tool | Description | Reference
 --- | --- | ---
 {% include tool.html id="Model validation" %} | Model Validation includes cross_validate, cross_val_predict, learning_curve, and more | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
-{% include tool.html id="CLF Metrics" %} | Calculate metrics for classification performance  | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
 {% include tool.html id="Pairwise Metrics" %} | Evaluate pairwise distances or compute affinity or kernel for sets of samples | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
-{% include tool.html id="Regression metrics" %} | Calculate metrics for regression performance | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
+{% include tool.html id="Train/Test evaluation" %} | Train, Test and Evaluation fit a model using part of dataset and evaluate using the rest | [Pedregosa et al. 2011](http://jmlr.csail.mit.edu/papers/v12/pedregosa11a.html){:target="_blank"}
```

Comment:
`Train, Test and Evaluation to fit a model ... `

Reply:
Similar comments for all other sentences containing names of tools and their descriptions